### PR TITLE
[Bot] Update inference types

### PR DESCRIPTION
**Pull request by HuggingFaceInfra**
This PR updates the inference types. It has been generated by running:
```sh
# Generate Python code
pnpm run inference-codegen  # from @huggingface/tasks-gen

# Copy to this repository
cp -r ./huggingface.js/packages/tasks/.python_generated/* ./huggingface_hub/src/huggingface_hub/inference/_generated/types

# Clean things
make inference_update
make style
```

This PR was automatically created by the [Update Inference Types workflow](https://github.com/huggingface/huggingface_hub/blob/main/.github/update-inference-types.yml).

Make sure the changes are correct before merging.

### CODE REVIEW

**Comment by Wauplin** [comment]
(generated as part of https://github.com/huggingface/huggingface_hub/pull/2600, let's not merge yet)

File: src/huggingface_hub/inference/_client.py [hunk 1]
```diff
@@ -2791,11 +2789,13 @@ def zero_shot_classification(
...
                 probabilities are normalized for each candidate.
             hypothesis_template (`str`, *optional*):
-                The sentence used in conjunction with candidateLabels to attempt the text classification by replacing
-                the placeholder with the candidate labels.
+                The sentence used in conjunction with `candidate_labels` to attempt the text classification by
+                replacing the placeholder with the candidate labels.
             model (`str`, *optional*):
                 The model to use for inference. Can be a model ID hosted on the Hugging Face Hub or a URL to a deployed
                 Inference Endpoint. This parameter overrides the model defined at the instance level. If not provided, the default recommended zero-shot classification model will be used.
+            candidate_labels (`List[str`, ):
```
Comment (hanouticelina) [inline src/huggingface_hub/inference/_client.py:2797]: i'm taking a look, something wrong with the auto-generated docstring!

File: src/huggingface_hub/inference/_client.py [hunk 2]
```diff
@@ -348,7 +348,6 @@ def audio_classification(
             top_k (`int`, *optional*):
                 When specified, limits the output to the top K most probable classes.
             function_to_apply (`"AudioClassificationOutputTransform"`, *optional*):
-                The function to apply to the output.
```
Comment (hanouticelina) [inline src/huggingface_hub/inference/_client.py:351]: I'll open a PR to add the docstring of `function_to_apply` in huggingface.js

Comment (hanouticelina) [reply src/huggingface_hub/inference/_client.py:351]: [huggingface/huggingface.js/pull/1043](https://github.com/huggingface/huggingface.js/pull/1043)

File: src/huggingface_hub/inference/_client.py (hunk 1 above)
Comment (hanouticelina) [reply src/huggingface_hub/inference/_client.py]: Fixed by #2668

File: src/huggingface_hub/inference/_generated/types/text_classification.py [hunk 3]
```diff
@@ -14,18 +14,14 @@
 
 @dataclass
 class TextClassificationParameters(BaseInferenceType):
-    """
-    Additional inference parameters for Text Classification.
+    """Additional inference parameters
+    Additional inference parameters for Text Classification
```
Comment (hanouticelina) [inline src/huggingface_hub/inference/_generated/types/text_classification.py:18]: (nit) the duplicate docstring comes from [huggingface/huggingface.js/tasks/text-classification/spec/input.json#L13](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/text-classification/spec/input.json#L13)  and [huggingface/huggingface.js/tasks/text-classification/spec/input.json#L20](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/text-classification/spec/input.json#L20), same for other tasks.

**Review by hanouticelina — APPROVED** [review]
looks good!

File: src/huggingface_hub/inference/_generated/types/text_classification.py (hunk 3 above)
Comment (Wauplin) [reply src/huggingface_hub/inference/_generated/types/text_classification.py:18]: Let's remove it then (we can still merge this PR first)

Comment (Wauplin) [reply src/huggingface_hub/inference/_generated/types/text_classification.py:18]:
```
		"parameters": {
			"$ref": "#/$defs/TextClassificationParameters"
		}
```
seems enough

Comment (hanouticelina) [reply src/huggingface_hub/inference/_generated/types/text_classification.py:18]: i'm opening a PR now to remove it!

**Review by hanouticelina — APPROVED** [review]

**Comment by hanouticelina** [comment]
the CI is finally 💚 let's merge this

**Pull request merged**